test/api/handlers_test.go

```diff
@@ -280,14 +280,53 @@ func testUserPUT(t *testing.T, at *test.AccountsTester) {
 		t.Fatalf("Expected the user to have StripeID %s, got %s", stripeID, u2.StripeID)
 	}
 	// Try to update the StripeID again. Expect this to fail.
-	r, _, err := at.UserPUT("", stripeID)
+	r, _, err := at.UserPUT("", "", stripeID)
 	if err == nil || !strings.Contains(err.Error(), "409 Conflict") || r.StatusCode != http.StatusConflict {
 		t.Fatalf("Expected to get error '409 Conflict' and status 409, got '%s' and %d", err, r.StatusCode)
 	}
 
+	// Update the user's password with an empty one. Expect this to succeed but
+	// not change anything.
+	_, b, err = at.UserPUT("", "", "")
+	if err != nil {
+		t.Fatal(err)
+	}
+	// Fetch the user from the DB again and make sure their password hash hasn't
+	// changed.
+	uSamePassHash, err := at.DB.UserByID(at.Ctx, u.ID)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if uSamePassHash.PasswordHash != u.PasswordHash {
```

Comment:
do we validate that the initial password is not empty?  So that `u.PasswordHash` isn't just a hash of an empty string, which in fact would be what was just set.

Reply:
Even if the old password was an empty string and the new one is also an empty string, their hashes would still be different because of the salt that's generated for each of them. So, if the password hash is the same we are sure that we didn't update the password, even if the old and new passwords are identical.